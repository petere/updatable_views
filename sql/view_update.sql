CREATE TABLE foo (a int, b text);

INSERT INTO foo VALUES (1, 'one');

CREATE VIEW foov1 AS SELECT a AS x, b AS y FROM foo WHERE a < 100;

CREATE TRIGGER foov1_update INSTEAD OF INSERT OR UPDATE OR DELETE ON foov1 FOR EACH ROW EXECUTE PROCEDURE update_view();

INSERT INTO foov1 VALUES (2, 'two');

SELECT * FROM foo;
SELECT * FROM foov1;
