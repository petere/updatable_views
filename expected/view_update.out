CREATE TABLE foo (a int, b text);
INSERT INTO foo VALUES (1, 'one');
CREATE VIEW foov1 AS SELECT a AS x, b AS y FROM foo WHERE a < 100;
CREATE TRIGGER foov1_update INSTEAD OF INSERT OR UPDATE OR DELETE ON foov1 FOR EACH ROW EXECUTE PROCEDURE update_view();
INSERT INTO foov1 VALUES (2, 'two');
NOTICE:  view INSERT
NOTICE:  do_insert = INSERT INTO foo (a, b) VALUES ($1, $2);
SELECT * FROM foo;
 a |  b  
---+-----
 1 | one
 2 | two
(2 rows)

SELECT * FROM foov1;
 x |  y  
---+-----
 1 | one
 2 | two
(2 rows)

